android/src/toga_android/window.py

```diff
@@ -1,47 +1,23 @@
+from .container import Container
 from .libs.android import R__id
 from .libs.android.view import ViewTreeObserver__OnGlobalLayoutListener
 
 
-class AndroidViewport(ViewTreeObserver__OnGlobalLayoutListener):
-    def __init__(self, native, container):
-        """
-        :param native: A native widget whose size will be tracked.
-        :param container: An object with a ``content`` attribute, which will have
-            ``refresh()`` called on it whenever the native widget's size changes.
-        """
+class LayoutListener(ViewTreeObserver__OnGlobalLayoutListener):
+    def __init__(self, window):
         super().__init__()
-        self.native = native
-        self.container = container
-        self.last_size = (None, None)
-        native.getViewTreeObserver().addOnGlobalLayoutListener(self)
-
-        self.dpi = native.getContext().getResources().getDisplayMetrics().densityDpi
-        # Toga needs to know how the current DPI compares to the platform default,
-        # which is 160: https://developer.android.com/training/multiscreen/screendensities
-        self.baseline_dpi = 160
-        self.scale = float(self.dpi) / self.baseline_dpi
+        self.window = window
 
     def onGlobalLayout(self):
         """This listener is run after each native layout pass.
 
         If any view's size or position has changed, the new values will be visible here.
         """
-        new_size = (self.width, self.height)
-        if self.last_size != new_size:
-            self.last_size = new_size
-            if self.container.content:
-                self.container.content.refresh()
-
-    @property
-    def width(self):
-        return self.native.getWidth()
+        native_parent = self.window.native_content.getParent()
+        self.window.resize_content(native_parent.getWidth(), native_parent.getHeight())
 
-    @property
-    def height(self):
-        return self.native.getHeight()
 
-
-class Window:
+class Window(Container):
```

Comment:
Is there a particular reason for using "is-A" rather than "has-A" as the composition for Window and Container? It's an interesting discrepancy between the iOS/Cocoa/GTK world and the Android one; I'm wondering what the source of that difference is.

Reply:
I agree the benefit of inheritance turned out to be fairly small. It's mainly just providing a single implementation of the `set_content` method, reducing the number of distinct objects you need to keep track of, and unifying the meaning of the word "container".

As discussed, my preference is to keep it the way it is for this PR, and revisit it in the SplitContainer PR where the container/content relationship will no longer be 1-1.